lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -429,11 +433,55 @@ function dosomething_user_validate_address_field($form, &$form_state) {
     }
     // We have a full address, save it!
     else {
-      dosomething_user_set_address_values($form, $form_state, $formatted_address);
+      if ($form['#form_id'] === 'dosomething_user_info_form') {
+        dosomething_user_set_address_values_from_info_form($form, $form_state, $formatted_address);
+      }
+      else {
+        dosomething_user_set_address_values_from_edit_form($form, $form_state, $formatted_address);
+      }
     }
```

Comment:
@angaither I'm not sure I quite understand how to use the menu hook to validate the way JS does so tried to go the second route you suggested of creating a new validation to handle the manually created fields. To create this new validation, I noticed that I would use a lot of the same code that is in `dosomething_user_validate_address_field` so I just created two new functions to set the different fields in the info vs. edit form and then send all to `dosomething_user_validate_address_field` to actually do the work of setting form errors and saving the fields. I _think_ this follows the route of creating a new validation but could be completely misunderstanding what you're saying?